ethersphere/swarm
# swarm, swap: pass chequebook address at start-up

This PR aims to complete [1548](https://github.com/ethersphere/swarm/issues/1548) and the duplicate (?) [1599](https://github.com/ethersphere/swarm/issues/1599). 

No unit tests were provided, as this functionality resides largely in `swarm.Start`. Creating an integration test was attempted, but this is not possible with a simulated backend. 

To test manually (I used Rinkeby):
1) set `swap.AllowedNetworkID` to 4
2) ask me for some test ethers (if you don't have them) on the account which you use to start Swarm 
3) start Swarm without provided `chequebook`: 
`swarm --swap --swap-backend-url https://rinkeby.infura.io/v3/<INFURA_KEY>`
4) start Swarm with provided `chequebook`: `swarm --swap --swap-backend-url https://rinkeby.infura.io/v3/<INFURA_KEY>` --chequebook <Check the log from step 3>

An Infura key can be created by logging in at infura.io. I can give mine as well in PM.

With regards to [1599](https://github.com/ethersphere/swarm/issues/1599): it seems to me that it is suggested here that we have to store the chequebook to disk, such that it can be automatically re-used the next time a node boots up. I would propose to create a separate issue for this.

- closes https://github.com/ethersphere/swarm/issues/1548
- closes https://github.com/ethersphere/swarm/issues/1599
